### PR TITLE
Renamed HACKING to Contributing.md and updated; added CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by KanishkKhurana**

### CODE REVIEW

**Comment by esabol** [comment]
@KanishkKhurana : I'm not sure the first 21 or 22 lines of CONTRIBUTING.md have any long-term usefulness to the project beyond Hacktoberfest. Please delete.

**Comment by esabol** [comment]
Also, the bulk of the text in CONTRIBUTING.md needs to be converted to Markdown.

**Comment by KanishkKhurana** [comment]
@esabol 
yes , i'll do this asap

**Comment by KanishkKhurana** [comment]
@esabol i have added markdown elements to the file. 
please let me know if there are any other changes. 
though i have only been coding for a few months , i get to learn a lot through these simple tasks and thus enjoy it !

**Comment by esabol** [comment]
Much better, @KanishkKhurana , but I think more Markdown changes need to be made to CONTRIBUTING.md.

If you look at how the file is rendered on GitHub, you see that some lines are word-wrapped that shouldn't be, like the if/then/else block, etc.

Check it out here:

https://github.com/gearman/gearmand/blob/e5784d8d6f29ee3ca92fd5c40f2ff4d220962298/CONTRIBUTING.md

**Comment by KanishkKhurana** [comment]
@esabol Please review the changes and do let me know if it needs more edits

**Comment by KanishkKhurana** [comment]
@esabol ,
i have made all the changes you have requested . 
please review them, sorry it took a while .
i request you to please add `hacktoberfest` and `hacktoberfest-accepted` labels to this PR as it will really help me .

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+`git clone https://github.com/gearman/gearmand`
+
+You will need to have git installed in order for this to work.
+
+If you are building from source control (i.e. git), you should setup your environment via: \
+`./bootstrap.sh autoreconf`
+
+This will build the files you will need in order to run `./configure`
+
+If you are working on the server, here are some handy environmental variables you can set so that you can debug make test:
```
Comment (esabol) [inline CONTRIBUTING.md:15]: Please add backticks around "make test".

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+ * GEARMAN_VALGRIND <-- runs the server under valgrind.
+ * GEARMAN_MANUAL_GDB <-- runs the server under a remote version of GDB.
+ * GEARMAN_LOG <-- generates a log file for you with the output of the debug for the server
+
+GDB will pause the server while you do run the GDB command.
```
Comment (esabol) [inline CONTRIBUTING.md:21]: Please add "Note that " to the beginning of this line and change "do run" to just "run".

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+If you are building from source control (i.e. git), you should setup your environment via: \
+`./bootstrap.sh autoreconf`
+
+This will build the files you will need in order to run `./configure`
+
+If you are working on the server, here are some handy environmental variables you can set so that you can debug make test:
+
+ * GEARMAN_VALGRIND <-- runs the server under valgrind.
+ * GEARMAN_MANUAL_GDB <-- runs the server under a remote version of GDB.
+ * GEARMAN_LOG <-- generates a log file for you with the output of the debug for the server
```
Comment (esabol) [inline CONTRIBUTING.md:19]: Please add a period at the end of this line.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+GDB will pause the server while you do run the GDB command.
+
+Coding Style
+------------
+
+1. Variables during assignment should be like: `a= 12;`
+
+2. When in doubt, use `()`. It means I clearly know that you meant for an operation to follow a specific order.
+
+3. Cast return types void when there is a possibility of failure (don't bother with printf, use common sense):  (void)some_function(...);
```
Comment (esabol) [inline CONTRIBUTING.md:30]: Please add backticks around "printf" and around "(void)some_function(...);".

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+7. Make sure structs have a typedef with a _st suffix, enums have a _t
+    suffix, and functions have a _fn suffix. For example:
+    ```
+    typedef struct gearman_task { ... } gearman_task_st;
+    typedef enum gearman_return { ... } gearman_return_t;
+    typedef gearman_return_t (gearman_complete_fn)(gearman_task_st *task);
+    ```
+
+
+Happy Hacking :octocat:\
```
Comment (esabol) [inline CONTRIBUTING.md:55]: You accidentally deleted the comma before the backslash. Please put it back.

**Review by esabol — CHANGES_REQUESTED** [review]
@KanishkKhurana: Just a few more minor changes and then we are done, I promise!

**Review by esabol — APPROVED** [review]

**Comment by esabol** [comment]
@KanishkKhurana: We were able to resolve our Travis CI issues with this repo. Now, you need to rebase your pull request in order to bring in the changes to the CI setup. Are you familiar with the steps for rebasing a PR? Let me know if you need some guidance on how to do that. Personally, I prefer the method described [here](https://anavarre.net/how-to-rebase-a-github-pull-request/).

**Comment by esabol** [comment]
> @KanishkKhurana: We were able to resolve our Travis CI issues with this repo. Now, you need to rebase your pull request in order to bring in the changes to the CI setup. Are you familiar with the steps for rebasing a PR? Let me know if you need some guidance on how to do that. Personally, I prefer the method described [here](https://anavarre.net/how-to-rebase-a-github-pull-request/).

Never mind! You don't need to do that. I was thinking of how GitLab CI/CD works. Just restarting the Travis CI jobs that failed before has worked, so I can merge your PR now.

Thanks, @KanishkKhurana !

**Comment by KanishkKhurana** [comment]
i had started reading the link you had given @esabol but looks like i got lucky . 
thank you !

**Pull request merged**